Uwaterloo 210J pfryerda edits

pfryerda opened the pull request:
Peter Fryer-Davis is making these edits as a part of Ashley Kelly's UWaterloo ENGL 210J class.

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -731,60 +731,60 @@ block style-property-name-dart-diff
 +makeExample('template-syntax/ts/app/app.component.html', 'event-binding-2')(format=".")
 :marked
-  Element events may be the more common targets, but Angular looks first to see if the name matches an event property
+  Element events may be the more common targets, but Angular looks first to see if the name matches the event property
```
I've liked all the changes up to here, but I think "an" is correct here.

pfryerda replied:
Does it match any event that matches the correct parameters?

pfryerda replied:
@kwalrath Thoughts on this?

kwalrath replied:
Yes, it matches any event property on the element, so I believe "an" is correct. "the" makes it sound like an element can have only one event property, or maybe that you're talking about a property named "event".

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -731,60 +731,60 @@ block style-property-name-dart-diff
 
 :marked
-  If the name fails to match an element event or an output property of a known directive,
+  If the name fails to match the element event or output property of a known directive,
```
the -> an

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -731,60 +731,60 @@ block style-property-name-dart-diff
 
   ### *$event* and event handling statements
-  In an event binding, Angular sets up an event handler for the target event.
+  In event binding, Angular sets up an event handler for the target event.
```
I'm not sure about this change. I'd consult the author.

pfryerda replied:
@kwalrath Who might that be?

kwalrath replied:
Looking at https://github.com/angular/angular.io/blob/master/public/docs/ts/latest/guide/template-syntax.jade, this file has had 23(!) contributors. Looking at "Blame", the author to blame for that line is... me. But only because I fixed some capitalization; it used to say "an Event Binding."

Looking around the web, the phrase "an event binding" is fairly common. It refers to a specific occurrence of event binding, where "event binding" refers to the practice in general.

So I'd switch back to "an event binding."

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -731,60 +731,60 @@ block style-property-name-dart-diff
 
   The binding conveys information about the event, including data values, through
-  an **event object named `$event`**.
+  an **event object named** `$event`.
```
Why move the `**` to disinclude the most important term?

pfryerda replied:
I did this because it's rare to see bolded code. Making it a code block already makes it stand out from the rest of the text, and it's rare that you will see a bolded piece of code. It's also not consistent with the rest of the document (at least the parts that I read).

kwalrath replied:
I see your point, but I'd still lean toward making the formatting reflect the phrasing, as long as the final result is still readable.

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -731,60 +731,60 @@ block style-property-name-dart-diff
 +makeExample('template-syntax/ts/app/app.component.html', 'without-NgModel')(format=".")
 :marked
-  We’re binding the input box `value` to a `firstName` property, and we’re listening for changes by binding to the input box’s `input` event.
+  Here, the input box `value` is bound to the `firstName` property. Changes are being listened for by binding to the input box’s `input` event.
```
Can you think of a way to reword this without the passive voice?

pfryerda replied:
How does this sounds:

> Here, the input box 'value' binds to the 'firstName' property. Listen for changes by binding to the input box's 'input' event.

kwalrath replied:
That's a good start, but the second sentence sounds like a command instead of a description. Maybe:

> This code sets the input box `value` property by binding to the `firstName` property. To listen for changes to the value, the code binds to the input box's `input` event.

(I wasn't sure about _this/the code_ as the actor, but I can't think of anything better. Another possibility was _template_, but that sounded too big/vague.)

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -731,60 +731,60 @@ block style-property-name-dart-diff
   When the user makes changes, the `input` event is raised, and the binding executes the statement within a context that includes the DOM event object, `$event`.
 
-  To update the `firstName` property, we must get the changed text by following
+  To update the `firstName` property, the changed text must be retrieved by following
```
Again, I like avoiding "we", but adding passive voice is generally bad.

pfryerda replied:
How about:

> To update the `firstName` property, the changed text is retrieved by following the path `$event.target.value`.

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -731,60 +731,60 @@ block style-property-name-dart-diff
   Directives typically raise custom events with an Angular [EventEmitter](../api/core/index/EventEmitter-class.html).
-  A directive creates an `EventEmitter` and exposes it as a property.
-  The directive calls `EventEmitter.emit(payload)` to fire an event, passing in a message payload that can be anything.
+  Directives create an `EventEmitter` and expose it as a property.
```
The original seems clearer.

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -731,60 +731,60 @@ block style-property-name-dart-diff
-  A directive creates an `EventEmitter` and exposes it as a property.
-  The directive calls `EventEmitter.emit(payload)` to fire an event, passing in a message payload that can be anything.
+  Directives create an `EventEmitter` and expose it as a property.
+  Directives call `EventEmitter.emit(payload)` to fire an event, passing in a message payload, which can be anything.
```
"The directive" seems clearer.

(But I like the "that" -> ", which" change.)

pfryerda replied:
Does changing both lines to say "The directive" make more sense in this situation then? I want to keep it consistent.

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -800,24 +800,23 @@ block style-property-name-dart-diff
   The component defines a `deleteRequest` property that returns an `EventEmitter`.
   When the user clicks *delete*, the component invokes the `delete()` method
-  which tells the `EventEmitter` to emit a `Hero` object.
+  telling the `EventEmitter` to emit a `Hero` object.
```
I have no opinion on "which tells" vs. "telling", but either one should have a comma before it.

pfryerda replied:
Agreed.

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -800,24 +800,23 @@ block style-property-name-dart-diff
 :marked
   When the `deleteRequest` event fires, Angular calls the parent component's `deleteHero` method,
-  passing the *hero-to-delete* (emitted by `HeroDetail`) in the `$event` variable.
+  passing the *hero-to-delete* (emitted by `HeroDetail`) into the `$event` variable.
```
"in" sounds more correct to me.

pfryerda replied:
Which is right?

- (in)  *Hero-to-delete* is a part of the '$event' variable.
- (into) The '$event' variable is assigned the value of *hero-to-delete*.

I don't know which of the two is right, but I thought it was the second one.

kwalrath replied:
It's the first one. The value of `$event` already _is_ the hero-to-delete (as set by the code `deleteRequest.emit(this.hero)`).

kwalrath commented:
I like most of your edits very much!

pfryerda commented:
Thanks! I've added further questions inline.

pfryerda commented:
@kwalrath I've made some of the changes that you mentioned earlier. Still waiting on the remaining comments for responses. Thanks!

kwalrath commented:
Sorry for the delay. Taking another look now...

kwalrath commented:
OK, all done. Could you make the final changes and resolve any conflicts? Thanks!

pfryerda commented:
@kwalrath, updated and rebased with master, but it's failing CI. Never used Travis before and I'm not familiar with it. Any chance you can help me debug?

kwalrath commented:
Last I heard, CI was in early days and failing it wasn't a showstopper.

So don't worry about it. :) I'll look at your changes now.

kwalrath commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -744,47 +744,46 @@ block style-property-name-dart-diff
-  A directive creates an `EventEmitter` and exposes it as a property.
-  The directive calls `EventEmitter.emit(payload)` to fire an event, passing in a message payload that can be anything.
+  The directive creates an `EventEmitter` and exposes it as a property.
+  The directives calls `EventEmitter.emit(payload)` to fire an event, passing in a message payload, which can be anything.
```
directives -> directive?

(subject-verb mismatch)

kwalrath commented:
LGTM aside from one tiny character, which I can fix.

Thanks very much for your work, @pfryerda!
